master/internal/rm/kubernetesrm/resource_pool.go

```diff
@@ -214,6 +216,9 @@ func (k *kubernetesResourcePool) receiveRequestMsg(ctx *actor.Context) error {
 			req := it.Value()
 			if req.AllocationRef == ref {
 				req.State = msg.State
+				if sproto.ScheduledStates[req.State] {
```

Comment:
one more issue I found testing this

after killing a job it will report twice the number of slots as we expect for a brief period until the job is no longer returned in the results.

This seems to happen because we get sent another ```UpdatePodStatus``` with the same state of ```SchedulingStateScheduled``` again.

We appear to have code preventing sending ```UpdatePodStatus``` with the same state already 
https://github.com/determined-ai/determined/blob/3f8bfee8b2558fad05380b5a8d15853e72c8793f/master/internal/rm/kubernetesrm/pods.go#L766

This code seems to have a bug where we never set ```containerIDToSchedulingState``` to anything other than ```SchedulingStateQueued``` in the initialization. I think we need to just set the state to fix this issue.

Reply:
See new changes